classes/package/package_version.php

```diff
@@ -50,6 +50,42 @@ class package_version {
      */
     public string $version;
 
+    /**
+     * @var bool package was uploaded by user
+     */
+    public bool $ismine;
+
+    /**
+     * @var bool package is provided by the application server
+     */
+    public bool $istrusted;
+
+    /**
+     * Constructs sql fragment used to retrieve package versions.
+     *
+     * @param string $where
+     * @param array $params
+     * @return array A list containing the constructed sql fragment and an array of parameters.
+     */
+    public static function sql_get(string $where = '', array $params = []): array {
```

Comment:
Im Grunde könntest du hier auch direkt den Query ausführen und die Ergebnisse casten, oder? Ich würde dann vorschlagen, dass du eine Funktion `get_many_where: array` und eine Funktion `get_where: ?package_version`  anbietest, die du dann in den anderen Funktionen aufrufst.